### ISSUE TITLE
Helm Find Files breaks on long filenames

### ISSUE THREAD

**Issue by herbertjones** [syl20bnr/spacemacs#3500]
Once I hit 71 characters in the entire path I get this error message:

Error running timer: (error "Lisp nesting exceeds `max-lisp-eval-depth'")

This leaves `Helm Find Files` in a broken state where the `Enter` key does nothing and the file list no longer updates.  I am not yet sure if this is an upstream issue.

I updated spacemacs and packages about 8 hours ago.

#### System Info
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.0
- Spacemacs branch: develop (rev. baff415)
- Distribution: spacemacs
- Layers:
```elisp
(emacs-lisp shell auto-completion git
            (ibuffer :variables ibuffer-group-buffers-by 'projects)
            org semantic
            (shell :variables shell-default-height 30 shell-default-position 'bottom)
            spell-checking syntax-checking version-control ycmd eyebrowse c-c++
            (haskell :variables haskell-enable-ghc-mod-support t haskell-enable-ghci-ng-support t haskell-enable-shm-support t)
            html javascript markdown php python racket elm)
```

**Comment by herbertjones** [comment]
Appears to affect any helm buffer.  For example, `SPC b b`, now hold any character for a few seconds.  The time it takes for the display to catch up to the input starts to lag until the error above appears.

#### System Info
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.0
- Spacemacs branch: develop (rev. 47ce170)
- Distribution: spacemacs
- Layers:
```elisp
nil```

**Comment by herbertjones** [comment]
If I delete the `elpa` directory and restart, the problem does not occur after the packages are downloaded and loaded.  However after restarting the problem will begin again.  I can only imagine some hook gets set when the helm packages are available on startup that isn't set when loading the packages at a later point.  This may be a spacemacs issue after all and not an upstream issue.

**Comment by herbertjones** [comment]
git bisect determined the following:

```
a5ac542ef16269a0b3bf144e1776ff5cbd9d11fd is the first bad commit
commit a5ac542ef16269a0b3bf144e1776ff5cbd9d11fd
Author: Eivind Fonn <evfonn@gmail.com>
Date:   Thu Oct 15 16:22:18 2015 +0200

    Add helm-flx
    
    See #3327

:040000 040000 9c0c21fe9d870b3918e9d6044c02cb2ad386fd11 d02c89fc62bce53d428a8fbf7deb5e35e088496d M      layers
```

**Comment by PythonNut** [comment]
@herbertjones @TheBB 

This is due to a [lack of dynamic programming](https://github.com/lewang/flx/issues/34) in `flx`. I dare say these cases are somewhat (?) unrepresentative of common workflows.

Adding DP is, unfortunately, a pretty hairy task when it comes to `flx`, so I'm not sure the timeframe required for me to add it.

**Comment by TheBB** [comment]
We don't necessarily have to fix flx... can we just disable it in the helm sessions where it's problematic, such as `helm-find-files`?

**Comment by PythonNut** [comment]
@TheBB would you like to file an issue over in my repo?

**Comment by TheBB** [comment]
I filed this: https://github.com/PythonNut/helm-flx/issues/9

I thought that was what brought you here, in fact. :-)

**Comment by PythonNut** [comment]
Nope, I got here via #3327, where I was mentioned. :)

**Comment by TheBB** [comment]
Should be fixed now in flx version 0.6 by the impressive https://github.com/lewang/flx/pull/78
